packages/wdio-devtools-service/src/auditor.ts

```diff
@@ -20,26 +23,92 @@ const SHARED_AUDIT_CONTEXT = {
     computedCache: new Map()
 }
 
+interface Audit {
+    audit: (opts: any, context: any) => Promise<any>,
+    defaultOptions: Record<string, any>
+}
+
+interface AuditResults {
+    'speed-index': MetricsResult
+    'first-contentful-paint': MetricsResult
+    'largest-contentful-paint': MetricsResult
+    'cumulative-layout-shift': MetricsResult
+    'total-blocking-time': MetricsResult
+    interactive: MetricsResult
+}
+
+interface AuditRef {
+    id: keyof AuditResults
+    weight: number
+}
+
+interface MainThreadWorkBreakdownResult {
+    details: {
+        items: {
+            group: string,
+            duration: number
+        }[]
+    }
+}
+
+interface DiagnosticsResults {
+    details: {
+        items: any[]
+    }
+}
+
+interface ResponseTimeResult {
+    numericValue: number
+}
+
+interface MetricsResult {
+    score: number
+}
+
+interface MetricsResults {
+    details: {
+        items: {
+            estimatedInputLatency: number
+            observedDomContentLoaded: number
+            observedFirstVisualChange: number
+            observedFirstPaint: number
+            firstContentfulPaint: number
+            firstMeaningfulPaint: number
+            largestContentfulPaint: number
+            observedLastVisualChange: number
+            firstCPUIdle: number
+            interactive: number
+            observedLoad: number
+            speedIndex: number
+            totalBlockingTime: number
+        }[]
+    }
+}
+
 export default class Auditor {
-    constructor(traceLogs, devtoolsLogs) {
-        this.devtoolsLogs = devtoolsLogs
-        this.traceLogs = traceLogs
+    _devtoolsLogs?: CDPSessionOnMessageObject[]
+    _traceLogs?: Trace
+    _url?: string
+
+    constructor(traceLogs?: Trace, devtoolsLogs?: CDPSessionOnMessageObject[]) {
```

Comment:
```suggestion
    constructor(private readonly traceLogs?: Trace, private readonly devtoolsLogs?: CDPSessionOnMessageObject[]) {
```

then remove the declarations above the constructor